controllers/dbaas_reconciler.go

```diff
@@ -262,20 +285,35 @@ func (r *DBaaSReconciler) checkInventory(inventoryRef v1alpha1.NamespacedName, D
 		return
 	}
 
-	// The inventory must be in ready status before we can move on
-	invCond := apimeta.FindStatusCondition(inventory.Status.Conditions, v1alpha1.DBaaSInventoryReadyType)
-	if invCond == nil || invCond.Status == metav1.ConditionFalse {
-		err = fmt.Errorf("inventory %v is not ready", inventoryRef)
-		logger.Error(err, "Inventory is not ready", "Inventory", inventory.Name, "Namespace", inventory.Namespace)
-		conditionFn(v1alpha1.DBaaSInventoryNotReady, v1alpha1.MsgInventoryNotReady)
-		if errCond := r.Client.Status().Update(ctx, DBaaSObject); errCond != nil {
-			if errors.IsConflict(errCond) {
-				logger.V(1).Info("DBaaS Object modified", "DBaaS Object", DBaaSObject)
-			} else {
-				logger.Error(errCond, "Error updating the DBaaS Object resource status", "DBaaS Object", DBaaSObject)
-			}
+	validNS, err = r.isValidConnectionNS(ctx, DBaaSObject.GetNamespace(), inventory)
+	if err != nil {
+		return inventory, validNS, err
+	}
+	if validNS {
+		// The inventory must be in ready status before we can move on
+		invCond := apimeta.FindStatusCondition(inventory.Status.Conditions, v1alpha1.DBaaSInventoryReadyType)
+		if invCond == nil || invCond.Status == metav1.ConditionFalse {
+			err = fmt.Errorf("inventory %v is not ready", inventoryRef)
+			logger.Error(err, "Inventory is not ready", "Inventory", inventory.Name, "Namespace", inventory.Namespace)
+			conditionFn(v1alpha1.DBaaSInventoryNotReady, v1alpha1.MsgInventoryNotReady)
+		}
```

Comment:
```suggestion
		} else {
		     return
		}
```
It is ok to return and not to update the status in this case.

Reply:
you mean instead of this? we need the following status set in connection/instance objects.
```go
} else {
		switch DBaaSObject.(type) {
		case *v1alpha1.DBaaSConnection:
			conditionFn(v1alpha1.DBaaSInvalidNamespace, v1alpha1.MsgConnInvalidNamespace)
		case *v1alpha1.DBaaSInstance:
			conditionFn(v1alpha1.DBaaSInvalidNamespace, v1alpha1.MsgConnInvalidNamespace)
```

Reply:
No. I mean if `validNS` is true and `invCond.Status` is true, the function can return directly without updating the status of the object.

Reply:
deon